pkg/controller/catalog/storage/mem_storage_instance.go

```diff
@@ -0,0 +1,63 @@
+package storage
+
+import (
+	"errors"
+
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
+)
+
+var (
+	errInstanceAlreadyExists = errors.New("instance already exists")
+	errNoSuchInstance        = errors.New("no such instance")
+)
+
+type memStorageInstance struct {
+	// maps instance ID to instance
+	instances map[string]*servicecatalog.Instance
+}
+
+func newMemStorageInstance() *memStorageInstance {
+	return &memStorageInstance{instances: make(map[string]*servicecatalog.Instance)}
+}
+
+func (m *memStorageInstance) List() ([]*servicecatalog.Instance, error) {
+	ret := make([]*servicecatalog.Instance, len(m.instances))
+	i := 0
+	for _, val := range m.instances {
+		ret[i] = val
+		i++
+	}
+	return ret, nil
+}
+
+func (m *memStorageInstance) Get(name string) (*servicecatalog.Instance, error) {
+	ret, ok := m.instances[name]
+	if !ok {
+		return nil, errNoSuchInstance
+	}
+	return ret, nil
```

Comment:
I don't have a strong opinion on the naming, but I do worry about people using this storage and seeing behavior that they shouldn't could on - like edits to a struct being manifested in the storage w/o going thru Update().  Sure they'll hopefully notice it not acting that way when they switch to etcd but I'd prefer if this storage were just a little more real.